src/sentry/tasks/statistical_detectors.py

```diff
@@ -335,73 +336,144 @@ def query_transactions_timeseries(
-                            params["start"],
-                            params["end"],
-                            interval,
-                            "time",
-                        ),
-                        "project": project_id,
-                        "order": item["order"],
-                    },
-                    params["start"],
-                    params["end"],
-                    interval,
+        transactions_condition = BooleanCondition(
+            BooleanOp.OR,
+            [
+                BooleanCondition(
+                    BooleanOp.AND,
+                    [
+                        Condition(Column("project_id"), Op.EQ, project_id),
+                        Condition(Column("transaction"), Op.EQ, transaction_name),
+                    ],
                 )
-                yield project_id, transaction_name, formatted_result
+                for project_id, transaction_name in transactions
+            ],
+        )
+        query = Query(
+            match=Entity(EntityKey.GenericMetricsDistributions.value),
+            select=[
+                Column("project_id"),
+                Function(
+                    "arrayElement",
+                    (
+                        CurriedFunction(
+                            "quantilesIf",
+                            [0.95],
+                            (
+                                Column("value"),
+                                Function("equals", (Column("metric_id"), duration_metric_id)),
+                            ),
+                        ),
+                        1,
+                    ),
+                    "p95_transaction_duration",
+                ),
+                Function(
+                    "transform",
+                    (
+                        Column(f"tags_raw[{transaction_name_metric_id}]"),
+                        Function("array", ("",)),
+                        Function("array", ("<< unparameterized >>",)),
+                    ),
+                    "transaction",
+                ),
+            ],
+            groupby=[
+                Column("transaction"),
+                Column("project_id"),
+                Function(
+                    "toStartOfInterval",
+                    (Column("timestamp"), Function("toIntervalSecond", (3600,)), "Universal"),
+                    "time",
+                ),
+            ],
+            where=[
+                Condition(Column("org_id"), Op.IN, list(org_ids)),
+                Condition(Column("project_id"), Op.IN, list(project_ids)),
+                Condition(Column("timestamp"), Op.GTE, start),
+                Condition(Column("timestamp"), Op.LT, end),
+                Condition(Column("metric_id"), Op.EQ, duration_metric_id),
+                transactions_condition,
+            ],
+            orderby=[
+                OrderBy(Column("project_id"), Direction.ASC),
+                OrderBy(Column("transaction"), Direction.DESC),
```

Comment:
out of curiosity why are we ordering transaction names in descending order and project ids in ascending?

Reply:
Thanks, I meant to ask Volo about that - he had `OrderBy(Column("transaction"), Direction.DESC),` in his version so I left it, but I don't know why. cc @volokluev

Reply:
no reason

Reply:
:+1: swapped to ASC for consistency